src/cljs/coffee_re_frame/engine.cljs

```diff
@@ -61,12 +65,15 @@
 ;; Handle timing, sometimes advance to next step
 (defn handle-tick [cofx _]
   (let [db (:db cofx)
+        seconds-remaining-in-step (::seconds-remaining-in-step cofx)
         new-db (-> db increment-ticks update-incremental-volume)]
     {:db new-db
-     :fx [(if (should-advance new-db) [:dispatch [::next-step]] nil)]}))
+     :fx [(if (should-advance new-db) [:dispatch [::next-step]] nil)
+          (if (should-vibrate seconds-remaining-in-step) [:vibrate 100] nil)]}))
```

Comment:
We conditionally add the `:vibrate` effect that we created earlier based on the result of the `should-vibrate` function